refactor(curriculum): JS comparisons quiz

zairahira opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #58296

<!-- Feel free to add any additional description of changes below this line -->

zairahira committed:
modify questions

moT01 reviewed (commented):
It's looking good @zairahira - here's what I wrote down...

2: I would change it so the code example is below the question text. E.g. "What is the output of the following code?" (followed by the code example)

 Same with 5, 10, 11, and 12, 13, 15, 18, and 20. This is the structure we decided on.  

7: Change to use a question mark at the end of the text.  

9: The answer with text that is that much longer than the other answers is often the correct one. Can we make the answers closer to the same length?  

10: I don't actually see that taught in this section. Perhaps it's in a previous section - or perhaps people can figure it out, cause it's closely related. Maybe we should find something else from this section?  

17 and 19: Feels like these are from an earlier section. We should maybe find something else here.

20: I'm not sure this was covered. 
 
Note that we added the ability to have quizzes only be 10 questions. Perhaps this would be a good candidate for that since it's such a short section. It would be nice to include one or two more with switch examples if you need to replace some questions. There wasn't many on those. Also, I noticed that some of those questions were discussed in the issue and maybe decided to be left alone. That's fine if you want.

zairahira committed:
fix code examples order

zairahira committed:
shorten correct answer to Q9

zairahira committed:
reduce redundant questions

zairahira committed:
Merge remote-tracking branch 'upstream/main' into chore/refactor-js-arrays-quiz

zairahira commented:
@moT01,

I have implemented the following suggestions:
- Removed redundant questions.
- Added one code example for `switch` statements.
- Modified the order where code should come below the question.

The remaining points of the original issue have been covered by removing redundant questions.

zairahira committed:
cleanup

Dario-DC reviewed (commented):
I think there should be some questions about comparisons with `null` and `undefined`.

The last question about this snippet should be replaced in my opinion:
```js
let x = 5;
if (x > 1 && x < 10) {
  console.log("x is between 1 and 10");
} else {
  console.log("x is not between 1 and 10");
}
```

zairahira committed:
Apply suggestions from code review

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

zairahira committed:
add backticks to switch

zairahira committed:
apply more suggestions

zairahira commented:
@Dario-DC,

I found this question was also repeated: 

```
What's the difference between `==` and `===`?`
```

I have replaced it with an `undefined/null` question.

As it is fine to have some repetition, I have left this question as it is:

```
Which logical operation does `||` represent?
```

moT01 reviewed (changes requested):
Sorry, couple more things @zairahira:

Question 5: Remove double quotes from the correct answer. 

Question 9: Looks weird because two answers are in single backticks, and two are in triple. Make them all triple. 

Question 10: Remove double quotes from the answers.  

Have we been using double quotes in places like that? Feels like we shouldn't be. If we've been using them, then we can just leave those.

zairahira committed:
apply suggestions from code review

zairahira committed:
typo

Dario-DC commented on curriculum/challenges/english/25-front-end-development/quiz-javascript-comparisons-and-conditionals/66edc47c11492ac5cf258ad9.md:
```diff
@@ -364,6 +181,8 @@ The code continues to evaluate the following case statements, even after finding
 
```
In the answer above `case` should be enclosed by backticks.

Dario-DC commented on curriculum/challenges/english/25-front-end-development/quiz-javascript-comparisons-and-conditionals/66edc47c11492ac5cf258ad9.md:
```diff
@@ -157,14 +109,14 @@ A value that is considered `true` or `false` when encountered in a Boolean conte
 
```
The distractors from the question above should have a starting `A` article.

zairahira committed:
apply suggestions
